src/sage/modular/drinfeld_modform/element.py

```diff
@@ -0,0 +1,397 @@
+            g1*g2 + g1^2
+            sage: (M.0 + M.1)*M.0
+            g1*g2 + g1^2
+        """
+        return self.__class__(self.parent(), self._polynomial*other._polynomial)
+
+    def _lmul_(self, c):
+        r"""
+        Return the scalar multiplication of self by `c`.
+
+        TESTS::
+
+            sage: A = GF(3)['T']; K = Frac(A); T = K.gen()
+            sage: M = DrinfeldModularForms(K, 2)
+            sage: (T^2 + T + 2) * M.0  # indirect doctest
+            (T^2 + T - 1)*g1
+            sage: M.1 * (T^5 + T^2)
+            (T^5 + T^2)*g2
+            sage: 0 * M.1
+            0
+            sage: M.0 * 0
+            0
+        """
+        return self.__class__(self.parent(), c*self._polynomial)
+
+    def __neg__(self):
+        r"""
+        Return the negation of self.
+
+        TESTS::
+
+            sage: A = GF(3)['T']; K = Frac(A)
+            sage: M = DrinfeldModularForms(K, 2)
+            sage: -M.0  # indirect doctest
+            -g1
+        """
+        return self.__class__(self.parent(), -self._polynomial)
+
+    def __bool__(self):
+        r"""
+        Return True whether self is nonzero.
+
+        TESTS::
+
+            sage: A = GF(3)['T']; K = Frac(A)
+            sage: M = DrinfeldModularForms(K, 2)
+            sage: bool(M.0)
+            True
+        """
+        return bool(self._polynomial)
+
+    def _latex_(self):
+        r"""
+        Return the latex expression of self.
+
+        TESTS::
+
+            sage: A = GF(3)['T']; K.<T> = Frac(A)
+            sage: M = DrinfeldModularForms(K, 2)
+            sage: M.inject_variables()
+            Defining g1, g2
+            sage: latex(g1)
+            g_{1}
+            sage: latex(g2)
+            g_{2}
+            sage: latex(1/T*g1^5 + g2*g1)
+            \frac{1}{T} g_{1}^{5} + g_{1} g_{2}
+        """
+        return self._polynomial._latex_()
+
+    def _richcmp_(self, other, op):
+        r"""
+        Return the comparison of self with other.
+
+        TESTS::
+
+            sage: A = GF(3)['T']; K = Frac(A)
+            sage: M = DrinfeldModularForms(K, 2)
+            sage: M.0 == M.1
+            False
+            sage: M.0 != M.1
+            True
+            sage: M.0 == M.0
+            True
+        """
+        if op != op_EQ and op != op_NE:
+            raise TypeError('invalid comparison between modular forms ring elements')
+        return richcmp(self._polynomial, other._polynomial, op)
+
+    def rank(self):
+        r"""
+        Return the rank of the graded Drinfeld form.
+
+        EXAMPLES::
+
+            sage: A = GF(3)['T']; K = Frac(A)
+            sage: M2 = DrinfeldModularForms(K, 2)
+            sage: (M2.0).rank()
+            2
+            sage: M5 = DrinfeldModularForms(K, 5)
+            sage: (M5.0 + M5.3).rank()
+            5
+        """
+        return self.parent()._rank
+
+    def is_one(self):
+        r"""
+        Return ``True`` whether the given graded Drinfeld form is the
```

Comment:
Here and subsequently you use "given". Sometimes you don't. You might want to uniformize the notation. Up to you !)